streaming/consumer.cc

```diff
@@ -36,6 +36,10 @@ reader_consumer_v2 make_streaming_consumer(sstring origin,
             auto cf = db.local().find_column_family(reader.schema()).shared_from_this();
             auto guard = service::topology_guard(frozen_guard);
             auto use_view_update_path = co_await db::view::check_needs_view_update_path(vb.local(), db.local().get_token_metadata_ptr(), *cf, reason);
+            // Disable data propagation to GSI when repairing the base table.
+            if (reason == streaming::stream_reason::repair && origin == sstables::repair_origin) {
+                use_view_update_path = false;
+            }
```

Comment:
This is probably a too heavy-handed fix here -- it completely disables generating view updates for repair.
This seems like just avoiding the problem, instead of fixing it.

Reply:
Yes. By tracking the code, I found a better solution. Please review it.

view_updates::generate_update → view_updates::update_entry → row::difference → collection_mutation::difference →  collection_mutation_view_description::difference →  compare_atomic_cell_for_merge → compare_unsigned

Actually compare_atomic_cell_for_merge  == 0 in https://github.com/scylladb/scylladb/blob/master/collection_mutation.cc#L353. As a result, diff is not assigned a value, so the cell is empty.

So the case where it is equal to 0 should be included.